assets/asset_manager.h

```diff
@@ -25,6 +25,8 @@ class AssetManager final : public AssetResolver {
 
   void PushBack(std::unique_ptr<AssetResolver> resolver);
 
+  void Merge(std::shared_ptr<AssetManager> assetManager);
```

Comment:
Since this is doing something fancier than a straight-up merge as I think is usually understood, it would be good to have a bunch of comments here about what this does.